src/shogun/distributions/classical/GaussianDistributionNew.cpp

```diff
@@ -0,0 +1,244 @@
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Written (W) 2013 Heiko Strathmann
+ */
+
+#include <shogun/distributions/classical/GaussianDistributionNew.h>
+
+#ifdef HAVE_EIGEN3
+
+#include <shogun/base/Parameter.h>
+#include <shogun/mathematics/eigen3.h>
+
+using namespace shogun;
+using namespace Eigen;
+
+CGaussianDistributionNew::CGaussianDistributionNew() : CProbabilityDistribution()
+{
+	init();
+}
+
+CGaussianDistributionNew::CGaussianDistributionNew(SGVector<float64_t> mean,
+		float64_t cov, bool cov_is_cholesky) :
+				CProbabilityDistribution(mean.vlen)
+{
+	init();
+
+	m_mean=mean;
+
+	m_cov=SGMatrix<float64_t>(1,1);
+	m_cov(1,1)=cov;
+}
+
+CGaussianDistributionNew::CGaussianDistributionNew(SGVector<float64_t> mean,
+		SGVector<float64_t> cov, bool cov_is_cholesky) :
+				CProbabilityDistribution(mean.vlen)
+{
+	REQUIRE(mean.vlen==cov.vlen, "Mean must have same dimension as "
```

Comment:
Maybe we can improve the error message a bit to
"Mean dimension (%d) and covariance dimension (%d) do not match"